### PR TITLE
SDA-8053 | Fix: make IDP password validation errors more readable

### PR DESCRIPTION
**Pull request by willkutler**
ref: https://issues.redhat.com/browse/SDA-8053?filter=-1

### CODE REVIEW

File: cmd/upgrade/cluster/cmd.go [hunk 1]
```diff
@@ -212,8 +212,7 @@ func createUpgradePolicyHypershift(r *rosa.Runtime, cmd *cobra.Command, clusterK
 	cluster *cmv1.Cluster, version string, scheduleDate string, scheduleTime string) error {
 	upgradePolicyBuilder := cmv1.NewControlPlaneUpgradePolicy().ScheduleType("manual").
 		UpgradeType("ControlPlane").Version(version)
-	// TODO: remove forceNow when we support scheduling
-	nextRun := buildUpgradeSchedule(r, cmd, scheduleDate, scheduleTime, true)
+	nextRun := buildUpgradeSchedule(r, cmd, scheduleDate, scheduleTime)
```
Comment (gdbranco) [inline cmd/upgrade/cluster/cmd.go:215]: Are these changes supposed to be included in this PR? Seems unrelated

File: cmd/create/idp/htpasswd.go [hunk 2]
```diff
@@ -187,19 +187,39 @@ func usernameValidator(val interface{}) error {
...
+		}
+		if tooShort {
+			pwdErrors = append(pwdErrors, fmt.Sprintf("must be at least 14 characters (got %d)", len(password)))
+		}
+		if len(pwdErrors) == 1 {
+			return fmt.Errorf("Password " + pwdErrors[0])
+		} else if len(pwdErrors) > 1 {
+			msg := "Password "
+			for i := 0; i < len(pwdErrors)-1; i++ {
+				msg += pwdErrors[i] + ", "
```
Comment (gdbranco) [inline cmd/create/idp/htpasswd.go:211]: Would it be possible to use strings.join instead and possibly drop the condition?

File: cmd/upgrade/cluster/cmd.go (hunk 1 above)
Comment (willkutler) [reply cmd/upgrade/cluster/cmd.go]: they aren't, just removed them

File: cmd/create/idp/htpasswd.go (hunk 2 above)
Comment (willkutler) [reply cmd/create/idp/htpasswd.go]: @gdbranco thanks for the review! I'd like to keep the condition since I think having the "and" before the last list item makes the message a bit more readable, but I switched the code to use `strings.Join` to make it cleaner.

File: cmd/create/idp/htpasswd.go [hunk 3]
```diff
@@ -187,19 +187,35 @@ func usernameValidator(val interface{}) error {
...
+			pwdErrors = append(pwdErrors, fmt.Sprintf("must be at least 14 characters (got %d)", len(password)))
+		}
 		if notAsciiOnly || containsSpace || tooShort {
-			return fmt.Errorf(
-				"password must be at least 14 characters (ASCII-standard) without whitespaces")
+			if len(pwdErrors) > 1 {
+				pwdErrors[len(pwdErrors)-1] = "and " + pwdErrors[len(pwdErrors)-1]
+			}
+
+			return fmt.Errorf("Password " + strings.Join(pwdErrors, ", "))
```
Comment (gdbranco) [inline cmd/create/idp/htpasswd.go:211]: Doesn't that outputs an extra ','?

Comment (gdbranco) [reply cmd/create/idp/htpasswd.go:211]:
```suggestion
			output := strings.Join(pwdErrors, ", ")
	                i := strings.LastIndex(output, ",")
	                if i != -1 {
		                output = output[:i] + " and" + output[i+len(","):]
	                }
```

**Comment by ciaranRoche** [comment]
Closing as solved by - https://github.com/openshift/rosa/pull/1475
